spec/ruby/optional/capi/string_spec.rb

```diff
@@ -1227,4 +1227,42 @@ def inspect
       -> { @s.rb_str_unlocktmp("test") }.should raise_error(RuntimeError, 'temporal unlocking already unlocked string')
     end
   end
+
+  describe "rb_enc_interned_str_cstr" do
+    it "returns a frozen string" do
+      str = "hello\0not used"
```

Comment:
**suggestion:** probably handling of `\0` byte isn't related to this test case. Otherwise I would expect it to be reflected in a test title:
```suggestion
      str = "hello"
```